src/Protocol/ActivityPub/Transmitter.php

```diff
@@ -675,6 +675,19 @@ private static function createPermissionBlockForItem(array $item, bool $blindcop
 
 		$exclusive = false;
 
+		if ($is_forum_thread) {
+			foreach (Tag::getByURIId($item['parent-uri-id'], [Tag::MENTION, Tag::EXCLUSIVE_MENTION]) as $term) {
+				$profile = APContact::getByURL($term['url'], false);
+				if (!empty($profile) && ($profile['type'] == 'Group')) {
+					if ($term['type'] == Tag::EXCLUSIVE_MENTION) {
+						$exclusive = true;
+					} elseif ($term['type'] == Tag::MENTION) {
+						$exclusive = false;
+					}
+				}
+			}
```

Comment:
This loop has a potential unpredictable outcome. If there are two mentions of a group, one exclusive and one not, the value of `$exclusive` will depend on which mention appears last in the array. Is this intended?

Reply:
Oups... Sry didn't saw it..

Reply:
Yeah, I guess that there is room for improvement.